### PR TITLE
feat(v2): enable JSX in markdown using MDX

### LINKED ISSUE #489: Embed React components in docs markdown?
```
# MyTite
Hello

## MySubtitle
Hello

<react>
    <MyComponent prop1={} prop2={}/>
</react>
```

### Is this a bug report?

No.

### Have you read the [Contributing Guidelines on issues]

Yes

### Environment

N/A

### Steps to Reproduce

N/A

### Expected Behavior

N/A

### Actual Behavior

N/A

### Reproducible Demo

N/A

### PR DESCRIPTION
**Pull request by endiliey**
## Motivation

Switch to MDX for v2. Might need more discussion since this is technically a move from `remarkable` to `mdx` as our markdown parser. Although i personally believe this shouldn't be a problem because we will allow user to choose their own markdown parser in v2 through plugins/ webpack config overriding.

Technically this is best done in v2 since v1 is not a SPA. So we will close #489

Other Changes:
- Had to fix the doc a little bit because some can be misclassified as JSX. Example: 
image
- Fix loader css

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Install some react component library at v2-website
```
yarn add react-trend react-youtube
```

Edit any doc file and go test it locally

```md
---
id: api-pages
title: JSX in Markdown
---

import YouTube from 'react-youtube';

## react-youtube

<YouTube videoId="dpw9EHDh2bM" />

## react-trend

import Trend from 'react-trend';

<Trend
  smooth
  autoDraw
  autoDrawDuration={3000}
  autoDrawEasing="ease-out"
  data={[0,2,5,9,5,10,3,5]}
  gradient={['#f72047', '#ffd200', '#1feaea']}
  radius={10}
  strokeWidth={2}
  strokeLinecap={'butt'}
/>
```

GIF Preview:
mdx

### CODE REVIEW

File: v2/lib/webpack/base.js [hunk 1]
```diff
@@ -105,10 +105,23 @@ module.exports = function createBaseConfig(props, isServer) {
...
+    {
+      test: /\.md$/,
+      use: [
+        'babel-loader',
+        '@mdx-js/loader',
+        'docusaurus/md-loader,
+      ]
+    }
+  */
   const mdRule = config.module.rule('markdown').test(/\.md$/);
```
Comment (yangshun) [inline v2/lib/webpack/base.js]: Do we want to include testing for `.mdx` here too?

**Comment by endiliey** [comment]
> On a side note, how does build size compare before and after now?
The build size (especially the js bundle) is definitely lower. I used https://www.npmjs.com/package/webpack-bundle-analyzer to check it before. 

Actually it's mostly caused due to removal of highlight.js which is known to have a very big bundle size. See https://bjacobel.com/2016/12/04/highlight-bundle-size/ or you can google it out

Ah, i also just realized that this PR removes syntax highlighting. Seems that MDX only support prismjs for now. Not much documentation on mdx itself on how to add syntax highlighting. Their examples are too focused on framework use like gatsby or next.js or even create-react-app. Had to dig out more ✌️ 

https://github.com/mdx-js/mdx/issues/253

**Comment by endiliey** [comment]
So I did some testing and find that mdx (even without syntax highlighting) creates bigger bundle size

MDX
after2

Remarkable
before2

There is also a problem of syntax highlighting. From my understanding, I've only seen mdx being compatible with prismjs.

**Comment by yangshun** [comment]
I think it's ok to go ahead and merge this even though there's a size increase. It's still the way to go. We can do some perf optimization later too.

**Pull request merged**